pkg/datapath/linux/config/config.go

```diff
@@ -1035,13 +1034,27 @@ func (h *HeaderfileWriter) writeStaticData(fw io.Writer, e datapath.EndpointConf
 func (h *HeaderfileWriter) WriteEndpointConfig(w io.Writer, e datapath.EndpointConfiguration) error {
 	fw := bufio.NewWriter(w)
 
+	nativeDevices, _ := tables.SelectedDevices(h.devices, h.db.ReadTxn())
+	deviceNames := tables.DeviceNames(nativeDevices)
+
+	// When WG & encrypt-node are on, a NodePort BPF to-be forwarded request
+	// to a remote node running a selected service endpoint must be encrypted.
+	// To make the NodePort's rev-{S,D}NAT translations to happen for a reply
+	// from the remote node, we need to attach bpf_host to the Cilium's WG
+	// netdev (otherwise, the WG netdev after decrypting the reply will pass
+	// it to the stack which drops the packet).
+	// In order to do so, we need to write the necessary static data here.
+	if option.Config.EnableNodePort && option.Config.EnableWireguard && option.Config.EncryptNode {
+		deviceNames = append(deviceNames, wgtypes.IfaceName)
+	}
+
 	writeIncludes(w)
-	h.writeStaticData(fw, e)
+	h.writeStaticData(deviceNames, fw, e)
 
-	return h.writeTemplateConfig(fw, e)
+	return h.writeTemplateConfig(nativeDevices, fw, e)
 }
 
-func (h *HeaderfileWriter) writeTemplateConfig(fw *bufio.Writer, e datapath.EndpointConfiguration) error {
+func (h *HeaderfileWriter) writeTemplateConfig(devices []*tables.Device, fw *bufio.Writer, e datapath.EndpointConfiguration) error {
```

Comment:
nit: Not a hard rule here but writeXXX with a writer would be best if writer figures as first arg.